internal/service/grid/distributionschedule_resource.go

```diff
@@ -0,0 +1,263 @@
+package grid
+
+import (
+	"context"
+	"fmt"
+	"net/http"
+
+	"github.com/hashicorp/terraform-plugin-framework/diag"
+	"github.com/hashicorp/terraform-plugin-framework/path"
+	"github.com/hashicorp/terraform-plugin-framework/resource"
+	"github.com/hashicorp/terraform-plugin-framework/resource/schema"
+	"github.com/hashicorp/terraform-plugin-framework/types"
+
+	niosclient "github.com/infobloxopen/infoblox-nios-go-client/client"
+
+	"github.com/infobloxopen/terraform-provider-nios/internal/flex"
+	"github.com/infobloxopen/terraform-provider-nios/internal/utils"
+)
+
+var readableAttributesForDistributionschedule = "active,start_time,time_zone,upgrade_groups"
+
+// Ensure provider defined types fully satisfy framework interfaces.
+var _ resource.Resource = &DistributionscheduleResource{}
+var _ resource.ResourceWithImportState = &DistributionscheduleResource{}
+
+func NewDistributionscheduleResource() resource.Resource {
+	return &DistributionscheduleResource{}
+}
+
+// DistributionscheduleResource defines the resource implementation.
+type DistributionscheduleResource struct {
+	client *niosclient.APIClient
+}
+
+func (r *DistributionscheduleResource) Metadata(ctx context.Context, req resource.MetadataRequest, resp *resource.MetadataResponse) {
+	resp.TypeName = req.ProviderTypeName + "_" + "grid_distributionschedule"
+}
+
+func (r *DistributionscheduleResource) Schema(ctx context.Context, req resource.SchemaRequest, resp *resource.SchemaResponse) {
+	resp.Schema = schema.Schema{
+		MarkdownDescription: "Manages a Distribution Schedule resource within the NIOS Grid.",
```

Comment:
```suggestion
		MarkdownDescription: "Manages a Distribution Schedule",
```
To be consistent with other objects